sleap_roots/pipelining.py

```diff
@@ -0,0 +1,625 @@
+            - tip_medianx: median value of x axis of all tip points
+            - tip_mediany: median value of y axis of all tip points
+            - tip_stdx: standard deviation of x axis of all tip points
+            - tip_stdy: standard deviation of y axis of all tip points
+            - tip_sdx_sdys: ratio of tip_stdx to tip_stdy
+            - tip_median_ratio: ratio of median value in all tip points to tip of primary root in y axis
+            - conv_perimeters: perimeter of convex hull
+            - conv_areas: area of convex hull
+            - conv_longest_dists: longest distance of convex hull
+            - conv_shortest_dists: shortest of convex hull
+            - conv_median_dists: median distance of convex hull
+            - conv_max_widths: maximum width of convex hull
+            - conv_max_heights: maxinum height of convex hull
+            - ellipse_a: semi-major axis length of the fitted ellipse
+            - ellipse_b: semi-minor axis length of the fitted ellipse
+            - ellipse_ratio: ratio of the minor to major lengths
+            - network_width_depth_ratio: width to depth ratio of bounding box for root
+            network
+            - network_solidity: total network length divided by the network convex area
+            - network_length_lower_network: root length in the lower fraction
+            - network_distribution_ratio: ratio of the root length in the lower fraction
+            over all root length
+            - count_scanline_interaction_max: maximum interaction of scanline and roots
+            - count_scanline_interaction_min: minimum interaction of scanline and roots
+            - count_scanline_interaction_mean: mean interaction of scanline and roots
+            - count_scanline_interaction_std: standard deviation interaction of scanline
+            and roots
+            - count_scanline_interaction_median: median interaction of scanline and
+            roots
+            - scanline_start: the scanline index where start interaction with roots
+            - scanline_end: the scanline index where end interaction with roots
+            - base_pctl: lateral root bases percentile in y-axis, # columns = # pctl
+            - tip_pctl: lateral root tips percentile in y-axis, # columns = # pctl
+    """
+    # check whether 2 *.slp files exist
+    plant_name = os.path.splitext(os.path.split(h5)[1])[0]
+    slp_files = [
+        file
+        for file in os.listdir(path)
+        if (file.endswith(".slp") and file.startswith(plant_name))
+    ]
+
+    if len(slp_files) < 2:
+        return "Incomplete SLEAP prediction!"
+
+    # get the primary and lateral roots points
+    # choose the longest primary root if more than one
+    pts_pr = get_pts_pr_lr(h5, rice=rice, frame=frame, primaryroot=True)
+    max_length_idx = np.nanargmax(get_root_lengths(pts_pr))
+    pts_pr = pts_pr[np.newaxis, max_length_idx]
+    pts_lr = get_pts_pr_lr(h5, rice=rice, frame=frame, primaryroot=False)
```

Comment:
Have these be an input to the function. We shouldn't be doing loading logic for each frame individually.

Basically the input here should be `pts_pr` and `pts_lr` (`pts_all` can be computed from those).

Reply:
Add functions (`get_primary_points`, `get_lateral_points`) in Series class.